### PR TITLE
Replace for...in by for(;;) when used on arrays

### PR DESCRIPTION
**Pull request by kubicle**
Fixes issue #34 
As a result WGo now works fine even if one added functions to Array's prototype.

Also in this PR: I saw 2 cases where "delete" was used on an array element (this leaves an "undefined" element in the array). It did not seem to be what was intended here, so I replaced by a "splice" but I might by wrong since I am new to this code and did not use/need these 2 methods myself. Please see my comments in the diff for the exact location of these.

### CODE REVIEW

**Comment by waltheri** [comment]
Yes, it is ok for me. Actually my code is quite messy, so everything would be ok :-) Thanks.

File: wgo/wgo.js [hunk 1]
```diff
@@ -1391,9 +1398,10 @@ Board.prototype = {
 	},
 	
 	removeCustomObject: function(handler, args) {
-		for(var key in this.obj_list) {
-			if(this.obj_list[key].handler == handler && this.obj_list[key].args == args) {
-				delete this.obj_list[key];
+		for(var i = 0; i < this.obj_list.length; i++) {
+			var obj = this.obj_list[i];
+			if(obj.handler == handler && obj.args == args) {
+				this.obj_list.splice(i, 1);
```
Comment (kubicle) [inline wgo/wgo.js]: See my comment in the PR for "splice" instead of "delete" here

File: wgo/wgo.js [hunk 2]
```diff
@@ -1417,10 +1425,11 @@ Board.prototype = {
...
 	removeEventListener: function(type, callback) {
-		for(var key in this.listeners) {
-			if(this.listeners[key].type == type && this.listeners[key].callback == callback) {
-				this.element.removeEventListener(this.listeners[key].type, this.listeners[key], true);
-				delete this.listeners[key];
+		for(var i = 0; i < this.listeners.length; i++) {
+			var listener = this.listeners[i];
+			if(listener.type == type && listener.callback == callback) {
+				this.element.removeEventListener(listener.type, listener, true);
+				this.listeners.splice(i, 1);
```
Comment (kubicle) [inline wgo/wgo.js]: See my comment in the PR for "splice" instead of "delete" here

**Comment by kubicle** [comment]
@waltheri hello, when you have time, this is ready for your review. Thanks!

**Comment by waltheri** [comment]
Thanks for the fix.

**Comment by yewang** [comment]
It seems that the "for ... in" pattern is still used in several other files. Was there a specific reason for not changing those as well?

**Comment by kubicle** [comment]
@waltheri you are welcome, thanks for WGo!

@yewang no specific reason, simply that I only used wgo.js (hence I am able to test it / understand it to some extent). Feel free to propose your changes in the other files. Thanks!

**Pull request merged**